resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1

```diff
@@ -584,6 +585,119 @@ class TextCursor
+        }
+    }
+
+    static [int] GetColor()
+    {
+        $FilterColorArgs = @{  Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::FilterColorProperty)}
+        if (-not(DoesRegistryKeyPropertyExist @FilterColorArgs))
+        {
+            return $false
+        }
+        else
+        {
+            $colorFilterSetting = (Get-ItemProperty @FilterColorArgs).FilterType
+            return $colorFilterSetting
+        }        
+    }
+
+    static [nullable[bool]] GetKeyboardShortcutStatus()
+    {
+        $KeyboardShortcutStatusArgs = @{ Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::HotkeyEnabledProperty); }
+        if (-not(DoesRegistryKeyPropertyExist @KeyboardShortcutStatusArgs))
+        {
+            return $false
+        }
+        else
+        {
+            $colorFilterSetting = (Get-ItemProperty @KeyboardShortcutStatusArgs).HotkeyEnabled
+            return $colorFilterSetting
+        }        
+    }
+
+    [ColorFilter] Get()
+    {
+        $currentState = [ColorFilter]::new()
+        $currentState.FilterEnabled = [ColorFilter]::GetStatus()
+        $currentState.FilterColor = [ColorFilter]::GetColor()
+        $currentState.HotkeyEnabled = [ColorFilter]::GetKeyboardShortcutStatus()
+        
+        return $currentState
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+        if (($null -ne $this.FilterEnabled) -and ($this.FilterEnabled -ne $currentState.FilterEnabled))
+        {
+            return $false
+        }
+        if (([ColorFilter]::KeepCurrentValue -ne $this.FilterColor) -and ($this.FilterColor -ne $currentState.FilterColor))
+        {
+            return $false
+        }
+        if (($null -ne $this.HotkeyEnabled) -and ($this.HotkeyEnabled -ne $currentState.HotkeyEnabled))
+        {
+            return $false
+        }
+
+        return $true
+    }
+
+    [void] Set()
+    {
+        if (-not $this.Test())
+        {
+            if ($null -ne $this.FilterEnabled) 
+            {
+                $FilterStatusArgs = @{ Path = $global:NTAccessibilityRegistryPath; Name = ([ColorFilter]::AccessibilityStatusKey); }
+                $ColorFilterValue = if ($this.FilterEnabled) { ([ColorFilter]::AccessibilityStatusValue) } else { "" }
+                Set-ItemProperty @FilterStatusArgs -Value $ColorFilterValue
+            }
+            
+            if ([PointerSize]::KeepCurrentValue -ne $this.FilterColor) 
+            {
+                $FilterColorArgs = @{  Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::FilterColorProperty)}
+                Set-ItemProperty @FilterColorArgs -Value $this.FilterColor -PropertyType DWord 
+            }
+            
+            if ($null -ne $this.HotkeyEnabled) 
+            {
+                $KeyboardShortcutStatusArgs = @{ Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::HotkeyEnabledProperty); }

```

Comment:
I would argue that in this scenario, creating a variable that you use once make it less readable. It makes sense in the Get() method above because you are reusing to check the existence of the key. But here it makes it more confusing and adds 2 unnecessary lines. 

`Set-ItemProperty -Path $global:ColorFilteringRegistryPath -Name [ColorFilter]::HotkeyEnabledProperty -Value $ColorFilterValue`

Please fix this and the other areas that do this.